File: src/spirite/pc/graphics/ImageBI.java
package spirite.pc.graphics;

import java.awt.image.BufferedImage;

import spirite.base.graphics.GraphicsContext;
import spirite.base.graphics.RawImage;
import spirite.pc.PCUtil;
import spirite.pc.graphics.awt.AWTContext;

public class ImageBI extends RawImage {
	public final BufferedImage img;

	public ImageBI( BufferedImage bi) {
		this.img = bi;
	}
	

	@Override
	public int getWidth() {
		return img.getWidth();
	}

	@Override
	public int getHeight() {
		return img.getHeight();
	}

	@Override
	public void flush() {
		img.flush();
	}

	@Override
	public int getByteSize() {
		return (img.getWidth()*img.getHeight()*img.getColorModel().getPixelSize() + 7)/8;
	}


	@Override
	public RawImage deepCopy() {
		return new ImageBI( PCUtil.deepCopy(img));
	}


	@Override
	public GraphicsContext getGraphics() {
		return new AWTContext( img.getGraphics(), img.getWidth(), img.getHeight());
	}


	@Override
	public int getRGB(int x, int y) {
		return img.getRGB(x, y);
	}


	@Override
	public boolean isGLOriented() {
		return false;
	}

}

File: src/spirite/pc/graphics/ImageBI.kt
package spirite.pc.graphics

import spirite.base.graphics.GraphicsContext
import spirite.base.graphics.RawImage
import spirite.base.util.Color

class ImageBI : RawImage {
    override fun getARGB(x: Int, y: Int): Int {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun getColor(x: Int, y: Int): Color {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override val graphics: GraphicsContext
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.

    override val width: Int
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
    override val height: Int
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.
    override val byteSize: Int
        get() = TODO("not implemented") //To change initializer of created properties use File | Settings | File Templates.

    override fun flush() {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }

    override fun deepCopy(): RawImage {
        TODO("not implemented") //To change body of created functions use File | Settings | File Templates.
    }
}